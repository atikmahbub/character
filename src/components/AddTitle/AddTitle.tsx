import { Typography, Box } from "@mui/material";
import ArrowBackIcon from "@mui/icons-material/ArrowBack";
import { Fragment } from "react";
import { useNavigate } from "react-router-dom";

type TitleProps = {
  title: string;
  backButton?: boolean;
};

const AddTitle = ({ title, backButton }: TitleProps) => {
  const navigate = useNavigate();
  return (
    <Fragment>
      <Typography variant="h3" textAlign="center">
        {title}
      </Typography>
      {backButton && (
        <Box
          color="#40bd24"
          display="flex"
          alignItems="center"
          justifyContent="center"
          gap={1}
          sx={{ cursor: "pointer" }}
          onClick={() => navigate(-1)}
        >
          <ArrowBackIcon />
          <Typography fontWeight={700} variant="caption" textAlign="center">
            Go Back
          </Typography>
        </Box>
      )}
    </Fragment>
  );
};

export default AddTitle;
