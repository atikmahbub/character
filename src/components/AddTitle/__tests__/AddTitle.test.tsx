import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import AddTitle from "../AddTitle";

const jsx = (
  <BrowserRouter>
    <AddTitle title="Hello" />
  </BrowserRouter>
);

test("Render Without BackButton", () => {
  const tree = renderer.create(jsx).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Render With BackButton", () => {
  const tree = renderer.create(jsx).toJSON();
  expect(tree).toMatchSnapshot();
});
