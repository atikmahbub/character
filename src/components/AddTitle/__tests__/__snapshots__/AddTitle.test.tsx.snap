// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render With BackButton 1`] = `
<h3
  className="MuiTypography-root MuiTypography-h3 css-47suzj-MuiTypography-root"
>
  Hello
</h3>
`;

exports[`Render Without BackButton 1`] = `
<h3
  className="MuiTypography-root MuiTypography-h3 css-47suzj-MuiTypography-root"
>
  Hello
</h3>
`;
